XamarinCommunityToolkit/Views/CameraView/CameraViewRenderer.uwp.cs

```diff
@@ -0,0 +1,397 @@
+					return max;
+				return (float)value;
+			}
+
+			var settings = new ZoomSettings
+			{
+				//TODO replace clamp
+				Value = Clamp(Element.Zoom, zoomControl.Min, zoomControl.Max),
+				Mode = zoomControl.SupportedModes.Contains(ZoomTransitionMode.Smooth)
+					? ZoomTransitionMode.Smooth
+					: zoomControl.SupportedModes.First()
+			};
+
+			zoomControl.Configure(settings);
+		}
+
+		DeviceInformation FilterCamera(DeviceInformationCollection cameraDevices, Windows.Devices.Enumeration.Panel panel)
+		{
+			foreach (var cam in cameraDevices)
+			{
+				if (cam.EnclosureLocation?.Panel == panel)
+					return cam;
+			}
+			return null;
+		}
+
+		async Task InitializeCameraAsync()
+		{
+			Available = false;
+			if (mediaCapture != null)
+				return;
+
+			var cameraDevices = await DeviceInformation.FindAllAsync(DeviceClass.VideoCapture);
+			if (cameraDevices.Count == 0)
+			{
+				Element?.RaiseMediaCaptureFailed("Camera devices not found.");
+				return;
+			}
+
+			IsBusy = true;
+			DeviceInformation device = null;
+			switch (Element.CameraOptions)
+			{
+				default:
+				case CameraOptions.Default:
+					device = cameraDevices[0];
+					break;
+				case CameraOptions.Front:
+					device = FilterCamera(cameraDevices, Windows.Devices.Enumeration.Panel.Front);
+					break;
+				case CameraOptions.Back:
+					device = FilterCamera(cameraDevices, Windows.Devices.Enumeration.Panel.Back);
+					break;
+				case CameraOptions.External:
+					device = FilterCamera(cameraDevices, Windows.Devices.Enumeration.Panel.Unknown);
+					break;
+			}
+			if (device == null)
+			{
+				Element?.RaiseMediaCaptureFailed($"{Element.CameraOptions} camera not found.");
+				IsBusy = false;
+				return;
+			}
+
+			var audioDevice = await DeviceInformation.FindAllAsync(DeviceClass.AudioCapture);
+			var selectedAudioDevice = audioDevice.Count == 0 ? null : audioDevice[0].Id;
+
+			mediaCapture = new MediaCapture();
+			try
+			{
+				await mediaCapture.InitializeAsync(new MediaCaptureInitializationSettings
+				{
+					VideoDeviceId = device.Id,
+					MediaCategory = MediaCategory.Media,
+					StreamingCaptureMode = selectedAudioDevice == null ? StreamingCaptureMode.Video : StreamingCaptureMode.AudioAndVideo,
+					AudioProcessing = Windows.Media.AudioProcessing.Default,
+					AudioDeviceId = selectedAudioDevice
+				});
+				flash = await Lamp.GetDefaultAsync();
+				if (mediaCapture?.VideoDeviceController.ZoomControl.Supported ?? false)
+					Element.MaxZoom = mediaCapture.VideoDeviceController.ZoomControl.Max;
+				displayRequest.RequestActive();
+				DisplayInformation.AutoRotationPreferences = DisplayOrientations.Landscape;
+			}
+			catch (UnauthorizedAccessException ex)
+			{
+				Element?.RaiseMediaCaptureFailed($"The app was denied access to the camera or microphone. {ex.Message}");
+				IsBusy = false;
+				return;
+			}
+
+			try
+			{
+				Control.Source = mediaCapture;
+				await mediaCapture.StartPreviewAsync();
```

Comment:
This causes a com exception for me when switching the camera type to fast.  The try block doesn't catch it either

Reply:
It's too late...that's because it's only it's not a file load exception

Reply:
added a catch for the com exception which raises a capture failed message in CameraView.

Reply:
Looks like this `FileLoadException` is a special one
https://docs.microsoft.com/en-us/windows/uwp/audio-video-camera/camera